zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/CassandraTagStore.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2015-2018 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.storage.cassandra.v1;
+
+import com.datastax.driver.core.Session;
+import java.util.List;
+import zipkin2.Call;
+import zipkin2.storage.TagStore;
+
+public class CassandraTagStore implements TagStore {
+  private final Session session;
+  private final Call<List<String>> tagKeys;
+  private final SelectTagValues.Factory tagValues;
+  CassandraTagStore(CassandraStorage storage){
+    this.session = storage.session.get();
+    tagKeys = new SelectTagKeys.Factory(session).create();
+    tagValues = new SelectTagValues.Factory(session);
+  }
+  @Override public Call<List<String>> getTagKeys() {
```

Comment:
no need if keys are immutable (collections.unmodifiable or util.sortedlist)

Reply:
Make this derived from a constant list from config I think.  maybe abstract base class instead of interface